src/devices/laserFromExternalPort/laserFromExternalPort.cpp

```diff
@@ -0,0 +1,303 @@
+    yInfo() << "LaserFromExternalPort closed";
+    return true;
+}
+
+bool LaserFromExternalPort::getDistanceRange(double& min, double& max)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    min = m_min_distance;
+    max = m_max_distance;
+    return true;
+}
+
+bool LaserFromExternalPort::setDistanceRange(double min, double max)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    m_min_distance = min;
+    m_max_distance = max;
+    return true;
+}
+
+bool LaserFromExternalPort::getScanLimits(double& min, double& max)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    min = m_min_angle;
+    max = m_max_angle;
+    return true;
+}
+
+bool LaserFromExternalPort::setScanLimits(double min, double max)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    yWarning("setScanLimits not yet implemented");
+    return true;
+}
+
+bool LaserFromExternalPort::getHorizontalResolution(double& step)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    step = m_resolution;
+    return true;
+}
+
+bool LaserFromExternalPort::setHorizontalResolution(double step)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    yWarning("setHorizontalResolution not yet implemented");
+    return true;
+}
+
+bool LaserFromExternalPort::getScanRate(double& rate)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    yWarning("getScanRate not yet implemented");
+    return true;
+}
+
+bool LaserFromExternalPort::setScanRate(double rate)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    yWarning("setScanRate not yet implemented");
+    return false;
+}
+
+
+bool LaserFromExternalPort::getRawData(yarp::sig::Vector &out)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    out = m_laser_data;
+    m_device_status = yarp::dev::IRangefinder2D::DEVICE_OK_IN_USE;
+    return true;
+}
+
+bool LaserFromExternalPort::getLaserMeasurement(std::vector<LaserMeasurementData> &data)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+#ifdef LASER_DEBUG
+        //yDebug("data: %s\n", laser_data.toString().c_str());
+#endif
+    size_t size = m_laser_data.size();
+    data.resize(size);
+    if (m_max_angle < m_min_angle) { yError() << "getLaserMeasurement failed"; return false; }
+    double laser_angle_of_view = m_max_angle - m_min_angle;
+    for (size_t i = 0; i < size; i++)
+    {
+        double angle = (i / double(size)*laser_angle_of_view + m_min_angle)* DEG2RAD;
+        data[i].set_polar(m_laser_data[i], angle);
+    }
+    m_device_status = yarp::dev::IRangefinder2D::DEVICE_OK_IN_USE;
+    return true;
+}
+bool LaserFromExternalPort::getDeviceStatus(Device_status &status)
+{
+    std::lock_guard<std::mutex> guard(mutex);
+    status = m_device_status;
+    return true;
+}
+
+bool LaserFromExternalPort::threadInit()
+{
+#ifdef LASER_DEBUG
+    yDebug("LaserFromExternalPort:: thread initialising...\n");
+    yDebug("... done!\n");
+#endif
+
+    return true;
+}
+
+void LaserFromExternalPort::run()
+{
+#ifdef DEBUG_TIMING
+    double t1 = yarp::os::Time::now();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 't1' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=4964370)